### PR TITLE
Latin17rgb pad use IS3731 IC

### PR DESCRIPTION
**Pull request by 18438880**
## Description

Latin17rgb pad use IS3731 IC

## Types of Changes

- [x] Core
- [x] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/latin17rgb/latin17RGB.json [hunk 1]
```diff
@@ -0,0 +1,118 @@
+  {
```
[the remark this answers is not in the document]
Comment (18438880) [reply keyboards/latin17rgb/latin17RGB.json:1]: THANKS A LOT sir.

**Review by drashna — APPROVED** [review]

File: keyboards/latin17rgb/readme.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Latin17RGB
+
+![Latin17rgb](https://github.com/18438880/latin17RGB-pad/blob/main/latin17RGB%20PAD.jpg)
```
Comment (fauxpark) [inline keyboards/latin17rgb/readme.md:3]:
```suggestion
![Latin17rgb](https://i.imgur.com/UoGWpw9l.jpg)
```

File: keyboards/latin17rgb/readme.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  RGB PAD use IS31FL3731 IC
+
+* Keyboard Maintainer: [18438880](https://github.com/18438880)
+* Hardware Availability: https://github.com/18438880/latin17RGB-pad/blob/main/latin17RGB%20PAD.jpg
```
Comment (fauxpark) [inline keyboards/latin17rgb/readme.md:8]:
```suggestion
* Hardware Availability: 
```

File: keyboards/latin17rgb/rules.mk [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/latin17rgb/rules.mk:12]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
```

File: keyboards/latin17rgb/rules.mk [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+BLUETOOTH_ENABLE = no          # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no              # Audio output on port C6
+FAUXCLICKY_ENABLE = no         # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no            # Enable support for HD44780 based LCDs
+
+NO_USB_STARTUP_CHECK = no      # Disable initialization only when usb is plugged in
+
+LAYOUTS = numpad_5x4
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = IS31FL3731
```
Comment (fauxpark) [inline keyboards/latin17rgb/rules.mk:39]:
```suggestion
BLUETOOTH_ENABLE = no          # Enable Bluetooth
AUDIO_ENABLE = no              # Audio output
NO_USB_STARTUP_CHECK = no      # Disable initialization only when usb is plugged in
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = IS31FL3731

LAYOUTS = numpad_5x4
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/latin17rgb/rules.mk [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite        # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes          # Mouse keys
+EXTRAKEY_ENABLE = yes          # Audio control and System control
+CONSOLE_ENABLE = yes            # Console for debug
```
Comment (noroadsleft) [inline keyboards/latin17rgb/rules.mk:13]:
```suggestion
CONSOLE_ENABLE = no            # Console for debug
```

File: keyboards/latin17rgb/rules.mk [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+COMMAND_ENABLE = no            # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no              # USB Nkey Rollover
```
Comment (noroadsleft) [inline keyboards/latin17rgb/rules.mk:18]:
```suggestion
NKRO_ENABLE = no               # USB Nkey Rollover
```

File: keyboards/latin17rgb/readme.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  RGB PAD use IS31FL3731 IC
+
+* Keyboard Maintainer: [18438880](https://github.com/18438880)
+* Hardware Availability: 
+
+Make example for this keyboard (after setting up your build environment):
+
+    make latin17rgb:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (noroadsleft) [inline keyboards/latin17rgb/readme.md:14]:
```suggestion
Flashing example for this keyboard:

    make latin17rgb:default:flash

To reset the board into bootloader mode, hold the key at the top left of the keyboard while connecting the USB cable (also erases persistent settings).

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/latin17rgb/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        _______, _______, _______, 
+        _______, _______, _______, _______,
+        _______, _______
+	),
+    [3] = LAYOUT_numpad_5x4( 
+	    _______, _______, _______, _______,
+	    _______, _______, _______, _______,
+        _______, _______, _______, 
+        _______, _______, _______, _______,
+        _______, _______
```
[the remark this answers is not in the document]
Comment (18438880) [reply keyboards/latin17rgb/keymaps/via/keymap.c:46]: sir ,my electronics is reeally this mode

File: keyboards/latin17rgb/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        KC_P4,   KC_P5,   KC_P6,  
+        KC_P1,   KC_P2,   KC_P3,  KC_PENT,  
+        KC_P0,   KC_PDOT
+    ),
+    [1] = LAYOUT_numpad_5x4( 
+	    RGB_TOG, RGB_MOD, KC_PMNS, _______,
+	    _______, _______, _______,KC_PGDN,
+        _______, _______, _______, 
+        _______, _______, _______, _______,
+        _______, _______
```
[the remark this answers is not in the document]
Comment (18438880) [reply keyboards/latin17rgb/keymaps/default/keymap.c]: Thanks a lot sir

File: keyboards/latin17rgb/latin17rgb.c [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+	
+};
+
+led_config_t g_led_config = {
+    {
+        { 0,      1,     2,     3},
+        { 4,      5,     6,     7},
+        { 8,      9,     10},
+        { 11,     12,    13,   14},
+        { 15,     16},
```
Comment (fauxpark) [inline keyboards/latin17rgb/latin17rgb.c:53]: This does not match the electrical matrix. Empty spots in the matrix should be filled with `NO_LED`.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/latin17rgb/latin17rgb.c (hunk 11 above)
Comment (18438880) [reply keyboards/latin17rgb/latin17rgb.c]: Thanks a lot Sir

File: keyboards/latin17rgb/latin17rgb.c [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+        {128, 16},{144, 16},{160,  0}
+    },
+    {
+        4, 4, 4, 1,
+        1, 1, 4,
+        1, 4, 4, 1,
+        1, 4, 4,
+        4, 4, 4
+    }
+};
```
Comment (noroadsleft) [inline keyboards/latin17rgb/latin17rgb.c:69]: This code block should stay as it was.

```suggestion
led_config_t g_led_config = {
    {
        { 0,      1,     2,     3},
        { 4,      5,     6,     7},
        { 8,      9,     10},
        { 11,     12,    13,   14},
        { 15,     16},
    },
    {
        { 48, 32},{ 48, 48},{ 64, 48},{ 80, 48},
        { 16, 32},{ 16, 48}, { 32, 48},{ 64, 64},
        { 80, 16},{ 64, 32},{ 80, 32},
        {112,  0},{128,  0},{144,  0},{160,  0},
        {128, 16},{144, 16}
    },
    {
        4, 4, 4, 1,
        1, 1, 4, 1,
        1, 4, 4,
        1, 4, 4, 4,
        4, 4
    }
};
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**